README.md

```diff
@@ -28,6 +28,84 @@ git submodule update --init --recursive
 python -m build
 ```
 
+## CLP IR Readers
+
+CLP IR Readers provide a convenient interface for CLP IR decoding and search methods.
+
+### CLPIRStreamReader
+
+- Read/decode any arbitrary CLP IR stream (as an instance of `IO[bytes]`).
+- Can be used as an iterator that returns each log event as a `LogEvent` object.
+- Can search target log events by giving a search query:
+  - Searching log events within a certain time range.
+  - Searching log messages that match certain wildcard queries.
+
+### CLPIRFileReader
+
+- Simple wrapper around CLPIRStreamHandler that calls `open` with a given local path.
+
+### Example Code: Using CLPIRFileReader to iterate and print log events
+```python
+from pathlib import Path
+from clp_ffi_py.readers import CLPIRFileReader
+
+with CLPIRFileReader(Path("example.clp.zst")) as clp_reader:
+    for log_event in clp_reader:
+        # Print the log message with its timestamp properly formatted.
+        print(log_event.get_formatted_message())
+```
+Each log event is represented by a `LogEvent` object, which offers methods to retrieve its
+underlying details, such as the timestamp and the log message. For more information, refer
+to the docstring of `LogEvent`.
```

Comment:
Can we do one of the following:
1. describe the methods available
2. link to the doc string
3. provide instructions for people who do not know where to find the doc string